File: 0530 Minimum Absolute Difference in BST/0530.cpp
class Solution {
public:
    void inorder(TreeNode* cur, vector<int>& v){
        if (cur->left) inorder(cur->left, v);
        v.push_back(cur->val);
        if (cur->right) inorder(cur->right, v);
    }

    int getMinimumDifference(TreeNode* root) {
        vector<int> v;
        inorder(root, v);
        
        int ans = INT_MAX;

        for (int i = 1; i < v.size(); i++){
            ans = min(ans, v[i] - v[i-1]);
        }

        return ans;
    }
};

File: 0530 Minimum Absolute Difference in BST/0530.py
class Solution:
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        v = []

        def inorder(cur):
            if cur.left: inorder(cur.left)
            v.append(cur.val)
            if cur.right: inorder(cur.right)

        inorder(root)

        ans = v[-1] - v[0]

        for i in range(1, len(v)):
            ans = min(ans, v[i] - v[i-1])

        return ans
